backend/typescript/server.ts

```diff
@@ -61,7 +65,45 @@ app.get("/delete-user-by-email/:email", async (req, res) => {
   await userService.deleteUserByEmail(req.params.email);
   res.status(204).send();
 });
-/** ************************************************************************ */
+
+const authService: IAuthService = new AuthService();
+
+app.get("/generate-token/:email/:password", async (req, res) => {
+  const token = await authService.generateToken(
+    req.params.email,
+    req.params.password,
+  );
+  res.status(200).json(token);
+});
+
+app.get("/is-authorized/:token/:role", async (req, res) => {
+  if (req.params.role !== "Admin" && req.params.role !== "User") {
```

Comment:
Possible improvement is to use enum for these

Reply:
I think role is typed here to be either "Admin" or "User" (correct me if I'm wrong) so we should be fine without the enums

Reply:
I believe we do have enums implemented for roles in the user model.  It shouldn't be too difficult to introduce those here. Currently I think in our startercode roles are only "Admin" and "User", but project teams could choose to have more roles.